cmd/cloud/deployment.go

```diff
@@ -728,13 +729,22 @@ func deploymentCreate(cmd *cobra.Command, _ []string, out io.Writer) error { //n
 	}
 	// request is to create from a file
 	if inputFile != "" {
-		requestedFlags := cmd.Flags().NFlag()
-		if requestedFlags > 1 {
-			// other flags were requested
+		// Collect all requested flags except --deployment-file and --wait
+		allowedFlags := map[string]bool{
+			"wait":            true,
+			"deployment-file": true,
+			"verbosity":       true,
+		}
+		var disallowedFlagSet bool
+		cmd.Flags().Visit(func(f *pflag.Flag) {
+			if !allowedFlags[f.Name] {
+				disallowedFlagSet = true
+			}
+		})
+		if disallowedFlagSet {
```

Comment:
Can we add unit tests around this logic, to keep it intact, since this feels like a very custom thing around the input file flag